bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/handler/MideaACHandler.java

```diff
@@ -0,0 +1,1454 @@
+            updateChannel(CHANNEL_ALTERNATE_TARGET_TEMPERATURE,
+                    new QuantityType<Temperature>(response.getAlternateTargetTemperature(), SIUnits.CELSIUS));
+            updateChannel(CHANNEL_INDOOR_TEMPERATURE,
+                    new QuantityType<Temperature>(response.getIndoorTemperature(), SIUnits.CELSIUS));
+            updateChannel(CHANNEL_OUTDOOR_TEMPERATURE,
+                    new QuantityType<Temperature>(response.getOutdoorTemperature(), SIUnits.CELSIUS));
+            updateChannel(CHANNEL_HUMIDITY, new DecimalType(response.getHumidity()));
+        }
+
+        /**
+         * Reads the inputStream byte array
+         * 
+         * @return byte array
+         */
+        public synchronized byte @Nullable [] read() {
+            byte[] bytes = new byte[512];
+            InputStream inputStream = this.inputStream;
+
+            try {
+                int len = inputStream.read(bytes);
+                if (len > 0) {
+                    logger.debug("Response received length: {} Thing:{}", len, thing.getUID());
+                    bytes = Arrays.copyOfRange(bytes, 0, len);
+                    return bytes;
+                }
+            } catch (IOException e) {
+                String message = e.getMessage();
+                logger.debug(" Byte read exception {}", message);
+            }
+            return null;
+        }
+
+        /**
+         * Writes the packet that will be sent to the device
+         * 
+         * @param buffer socket writer
+         * @throws IOException writer could be null
+         */
+        public synchronized void write(byte[] buffer) throws IOException {
+            DataOutputStream writer = this.writer;
+
+            try {
+                writer.write(buffer, 0, buffer.length);
+            } catch (IOException e) {
+                String message = e.getMessage();
+                logger.debug("Write error {}", message);
+            }
+        }
+
+        /**
+         * Periodical polling. Thirty seconds minimum
+         */
+        private void scheduleConnectionMonitorJob() {
+            if (connectionMonitorJob == null) {
+                logger.debug("Starting connection monitor job in {} seconds for {} at {} after 30 second delay",
+                        config.pollingTime, thing.getUID(), ipAddress);
+                long frequency = config.pollingTime;
+                long delay = 30L;
+                connectionMonitorJob = scheduler.scheduleWithFixedDelay(connectionMonitorRunnable, delay, frequency,
+                        TimeUnit.SECONDS);
+            }
+        }
+
+        private void cancelConnectionMonitorJob() {
+            if (connectionMonitorJob != null) {
+                connectionMonitorJob.cancel(true);
+                logger.debug("Cancelling connection monitor job for {} at {}", thing.getUID(), ipAddress);
+                connectionMonitorJob = null;
```

Comment:
Introduce a local var and this null warning can be fixed too.
```suggestion
            ScheduledFuture<?> connectionMonitorJob = this.connectionMonitorJob;
            if (connectionMonitorJob != null) {
                connectionMonitorJob.cancel(true);
                logger.debug("Cancelling connection monitor job for {} at {}", thing.getUID(), ipAddress);
                this.connectionMonitorJob = null;
```

Reply:
Also did this but had an issue the connectionManagerJob not getting assigned null.  I did two things and it is now working, but haven't sorted out which one or both. Added this. and also a method
```
private void cancelConnectionMonitorJob() {
            ScheduledFuture<?> connectionMonitorJob = this.connectionMonitorJob;
            if (connectionMonitorJob != null) {
                connectionMonitorJob.cancel(true);
                logger.debug("Cancelling connection monitor job for {} at {}", thing.getUID(), ipAddress);
                this.connectionMonitorJob = null;
                resetConnectionManager();
            }
        }
```
```
        /**
         * Ensure Connection Manager is null
         */
        public void resetConnectionManager() {
            connectionMonitorJob = null;
        }
```

Reply:
Hmm dont fully understand as `this.connectionMonitorJob = null` would make sure it is always null at the end of the method.